datajunction-server/datajunction_server/sql/dag.py

```diff
@@ -1,22 +1,244 @@
 """
 DAG related functions.
 """
-import collections
 import itertools
-from typing import Deque, Dict, List, Optional, Set, Tuple, Union
+from typing import Dict, List, Optional, Set, Union
 
-from sqlmodel import Session, select
+from sqlalchemy import and_, func, literal, or_
+from sqlalchemy.orm import aliased, joinedload
+from sqlalchemy.sql.operators import is_
+from sqlmodel import Session, join, select
 
-from datajunction_server.models import Column
+from datajunction_server.models import AttributeType, Column, ColumnAttribute
 from datajunction_server.models.base import NodeColumns
-from datajunction_server.models.node import DimensionAttributeOutput, Node, NodeRevision
+from datajunction_server.models.node import (
+    DimensionAttributeOutput,
+    Node,
+    NodeRelationship,
+    NodeRevision,
+)
 from datajunction_server.models.node_type import NodeType
 from datajunction_server.utils import SEPARATOR, get_settings
 
 settings = get_settings()
 
 
+def get_dimensions_dag(
+    session: Session,
+    node_revision: NodeRevision,
+    attributes: bool = True,
```

Comment:
nit: `with_attributes` name would better match the `bool` value.

Reply:
Good point! Changing this